executors/cpp/coll.cpp

```diff
@@ -204,89 +179,91 @@ const string TestCollator(json_object *json_in) {
       const bool ignore_punctuation_bool = json_object_get_boolean(ignore_obj);
       if (ignore_punctuation_bool) {
         uni_coll->setAttribute(UCOL_ALTERNATE_HANDLING, UCOL_SHIFTED, status);
-        if (U_FAILURE(status)) {
-          test_result = error_message;
-          json_object_object_add(
-              return_json,
-              "error", json_object_new_string("error setAttribute"));
-          no_error = false;
-          cout << "# Error in setAttribute: " <<
-              label_string << " : " <<
-              test_result << endl;
+        if (check_icu_error(
+                status, return_json,
+                "set UCOL_ALTERNATE_HANDLING to UCOL_SHIFTED")) {
+          return json_object_to_json_string(return_json);
         }
       }
     }
 
     // Just to check the result.
     uni_coll->getAttribute(UCOL_ALTERNATE_HANDLING, status);  // ignore return
+    if (check_icu_error(
+            status, return_json,
+            "getet UCOL_ALTERNATE_HANDLING")) {
+      return json_object_to_json_string(return_json);
+    }
 
     // Perform the string comparison
     uni_result = uni_coll->compare(us1, us2, status);
-    if (U_FAILURE(status)) {
-        json_object_object_add(
-            return_json,
-            "error", json_object_new_string("error in uni_coll_compare"));
-      no_error = false;
-      cout << "## Error in uni_coll->compare: " <<
-          label_string << " : " <<
-          error_message << endl;
+    if (check_icu_error( status, return_json, "uni_coll_compare")) {
+      return json_object_to_json_string(return_json);
     }
+
     if (uni_coll) {
       uni_coll->getAttribute(UCOL_ALTERNATE_HANDLING, status);  // ignore result
     }
     delete uni_coll;
+    if (check_icu_error( status, return_json, "uni_coll->getATTRIBUTE")) {
+      return json_object_to_json_string(return_json);
+    }
   }
 
-  if (no_error) {
-    if (uni_result == UCOL_GREATER) {
-      coll_result = false;
+  if (uni_result == UCOL_GREATER) {
+    coll_result = false;
 
+    if (debug) {
       cout << "# UNI_RESULT: " << label_string << " " << uni_result <<
           "  s1: " << string1 << " s2: " << string2 << endl;
+    }
 
-      // Check unescaped versions.
-      char char_out1[1000] = "";
-      char char_out2[1000] = "";
-      us1.extract(char_out1, 1000, nullptr, status);  // ignore result
-      if (U_FAILURE(status)) {
-        test_result = error_message;
-        json_object_object_add(
-            return_json,
-            "error", json_object_new_string("error extracting us1"));
+    // Check unescaped versions.
+    char char_out1[1000] = "";
+    char char_out2[1000] = "";
+    us1.extract(char_out1, 1000, nullptr, status);  // ignore result
```

Comment:
Any potential for `us1.extract(...)` and `us2.extract(...)` to use the `UnicodeString::toUTF8String(std::string)` API instead?

Reply:
And in other files, too!